### ISSUE TITLE
Indentation style and consistency (for fishscript)

### ISSUE THREAD

**Issue by dag** [fish-shell/fish-shell#749]
In the commandline editor, `case` is indented.  With `fish_indent`, it's not:

```
 > switch cat
       case cat
           echo cute
   end
cute
 > echo 'switch cat; case cat; echo cute; end' | fish_indent
switch cat
case cat
    echo cute
end⏎
```

For automatic indentation in _vim-fish_ I use the former style, for consistency with what you get when typing in the shell, but I also support using `fish_indent` as a formatting program.

I'd like it if we could decide on a single preferred style, "enforce" it consistently in fish, and then I can do the same in vim-fish.  Perhaps we should also use that style consistently in the fish sources, but that's a lesser concern for me.  Perhaps we could just run `fish_indent` on the source scripts, but I don't know if that's completely safe or if it will result in worse formatting in some cases.

I'm fine with either style; they're about equally ugly to me. :wink:

Another difference is that the commandline uses four spaces and `fish_indent` uses tabs.  For vim-fish, tabs are good because it means you can use whatever style you prefer and just run `:retab` after formatting with `fish_indent`, so I'd like it to keep using tabs.  Not sure making the commandline use tabs too for consistency is worth it here though; tabs being 8 characters wide in terminals makes for rather ugly indentation I think.

Last, I'm wondering what do to with line continuations.  `fish_indent` simply joins them (seemingly with no regard for line length) and the commandline editor just continues on the next line with no indentation.  I'm not really happy with either: the whole reason for using line continuations is usually to cut down on line length, but not indenting means continued lines aren't visually distinct.  I think continued lines should be indented two or three levels more than the first line that is continued, as in:

```
begin
    echo foo bar \
            abc def \
            123 456
end
```

Or something like that.

**Comment by faho** [comment]
This case has been fixed, and CONTRIBUTING.md talks about fish_indent, which should be the one true source for style info.

**Comment by krader1961** [comment]
To clarify @faho's closing comment: I got tired of the inconsistent style of the scripts in the project and restyled them all in commit 11a60c83 using `fish_indent`. In commit 42eaacd3 I modified `make style-all` to also run those scripts through ` fish_indent`. The fish_indent program still has at least one formatting behavior that we want to change (see issue #2646) but it is nonetheless the authoritative source for how our scripts are to be formatted (like `gofmt` is for Go programs).
